diff --git libs/assimp-3.1.1/CMakeLists.txt libs/assimp-3.1.1/CMakeLists.txt
index d332428..1af59dc 100644
--- libs/assimp-3.1.1/CMakeLists.txt
+++ libs/assimp-3.1.1/CMakeLists.txt
@@ -76,7 +76,7 @@ SET( ASSIMP_INCLUDE_INSTALL_DIR "include" CACHE PATH
 	"Path the header files are installed to." )
 SET( ASSIMP_BIN_INSTALL_DIR "bin" CACHE PATH
 	"Path the tool executables are installed to." )
-SET ( ASSIMP_BUILD_STATIC_LIB OFF CACHE BOOL
+SET ( ASSIMP_BUILD_STATIC_LIB ON CACHE BOOL
     "Build a static (.a) version of the library" )
 
 SET(ASSIMP_DEBUG_POSTFIX "d" CACHE STRING "Debug Postfitx for lib, samples and tools")
diff --git libs/assimp-3.1.1/code/BlenderDNA.cpp libs/assimp-3.1.1/code/BlenderDNA.cpp
index 16e639e..53e9706 100644
--- libs/assimp-3.1.1/code/BlenderDNA.cpp
+++ libs/assimp-3.1.1/code/BlenderDNA.cpp
@@ -55,7 +55,7 @@ using namespace Assimp::Formatter;
 
 #define for_each BOOST_FOREACH
 bool match4(StreamReaderAny& stream, const char* string) {
-	char tmp[] = { 
+	int8_t tmp[] = { 
 		(stream).GetI1(), 
 		(stream).GetI1(),  
 		(stream).GetI1(), 
@@ -344,10 +344,10 @@ void SectionParser :: Next()
 	stream.SetCurrentPos(current.start + current.size);
 
 	const char tmp[] = {
-		stream.GetI1(),
-		stream.GetI1(),
-		stream.GetI1(),
-		stream.GetI1()
+		(char) stream.GetI1(),
+		(char) stream.GetI1(),
+		(char) stream.GetI1(),
+		(char) stream.GetI1()
 	};
 	current.id = std::string(tmp,tmp[3]?4:tmp[2]?3:tmp[1]?2:1);
 
