--- catkin_ws/src/bfl/CMakeLists.txt
+++ catkin_ws/src/bfl/CMakeLists.txt
@@ -54,7 +54,7 @@
 # Shared or static library
 # ------------------------
 IF (NOT LIBRARY_TYPE)
-  SET( LIBRARY_TYPE shared CACHE STRING "What type of library to build: (shared, static or both)")
+  SET( LIBRARY_TYPE static CACHE STRING "What type of library to build: (shared, static or both)")
 ENDIF (NOT LIBRARY_TYPE)
 
 
