--- catkin_ws/src/image_common/camera_info_manager/CMakeLists.txt
+++ catkin_ws/src/image_common/camera_info_manager/CMakeLists.txt
@@ -25,7 +25,7 @@ install(DIRECTORY include/${PROJECT_NAME}/
         DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION}
 )
 
-if(CATKIN_ENABLE_TESTING)
+if(CATKIN_ENABLE_TESTING AND NOT ANDROID)
   find_package(rostest)
   # Unit test uses gtest, but needs rostest to create a ROS environment.
   # Hence, it must be created as a normal executable, not using
