--- libs/collada-dom-2.4.0/CMakeLists.txt
+++ libs/collada-dom-2.4.0/CMakeLists.txt
@@ -109,6 +109,8 @@ include(CheckCXXSourceCompiles)
 include(CheckTypeSize)
 find_package(PkgConfig)
 
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++03")
+
 if( CMAKE_COMPILER_IS_GNUCC OR CMAKE_COMPILER_IS_GNUCXX )
   add_definitions("-fno-strict-aliasing -Wall")
 endif()

--- libs/collada-dom-2.4.0/dom/CMakeLists.txt
+++ libs/collada-dom-2.4.0/dom/CMakeLists.txt
@@ -27,7 +27,7 @@ if( OPT_COLLADA14 )
   install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/include/1.4 DESTINATION ${COLLADA_DOM_INCLUDE_INSTALL_DIR} COMPONENT ${COMPONENT_PREFIX}-dev  PATTERN ".svn" EXCLUDE PATTERN ".~" EXCLUDE)
 endif()
 
-add_library(collada-dom SHARED ${COLLADA_BASE_SOURCES})
+add_library(collada-dom ${COLLADA_BASE_SOURCES})
 target_link_libraries(collada-dom ${COLLADA_LIBS})
 set_target_properties(collada-dom PROPERTIES
   COMPILE_FLAGS "${COLLADA_COMPILE_FLAGS}"

--- libs/collada-dom-2.4.0/dom/external-libs/pcre-8.02/pcrecpp.cc
+++ libs/collada-dom-2.4.0/dom/external-libs/pcre-8.02/pcrecpp.cc
@@ -47,6 +47,7 @@
 #include "pcrecpp.h"
 #include "pcre_stringpiece.h"
 
+#undef HAVE_STRTOQ
 
 namespace pcrecpp {
