diff --git a/image_pipeline/image_publisher/CMakeLists.txt b/image_pipeline/image_publisher/CMakeLists.txt
index b17dec9..55445ff 100644
--- a/image_pipeline/image_publisher/CMakeLists.txt
+++ b/image_pipeline/image_publisher/CMakeLists.txt
@@ -13,15 +13,32 @@ else()
   find_package(OpenCV 3 REQUIRED COMPONENTS ${opencv_3_components})  
 endif()
 
+include_directories(${catkin_INCLUDE_DIRS})
+
+find_package(PkgConfig)
+
+if (ANDROID)
+    find_package(yaml-cpp)
+    add_definitions(-DHAVE_NEW_YAMLCPP)
+else()
+  pkg_check_modules(YAML_CPP yaml-cpp)
+  if(${YAML_CPP_VERSION} VERSION_GREATER 0.5)
+    add_definitions(-DHAVE_NEW_YAMLCPP)
+  endif()
+  link_directories(${YAML_CPP_LIBRARY_DIRS})
+endif()
+include_directories(${YAML_CPP_INCLUDE_DIRS})
+
 # generate the dynamic_reconfigure config file
 generate_dynamic_reconfigure_options(cfg/ImagePublisher.cfg)
 
-catkin_package()
-
-include_directories(${catkin_INCLUDE_DIRS})
+catkin_package(
+  DEPENDS YAML_CPP OpenCV
+  CATKIN_DEPENDS cv_bridge dynamic_reconfigure camera_info_manager sensor_msgs image_transport nodelet roscpp
+)
 
-add_library(${PROJECT_NAME} SHARED src/nodelet/image_publisher_nodelet.cpp)
-target_link_libraries(${PROJECT_NAME} ${catkin_LIBRARIES} ${OpenCV_LIBRARIES})
+add_library(${PROJECT_NAME} src/nodelet/image_publisher_nodelet.cpp)
+target_link_libraries(${PROJECT_NAME} ${catkin_LIBRARIES} ${OpenCV_LIBRARIES} ${YAML_CPP_LIBRARIES})
 add_dependencies(${PROJECT_NAME} ${PROJECT_NAME}_gencfg)
 install(TARGETS image_publisher
         DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
@@ -29,7 +46,7 @@ install(TARGETS image_publisher
 
 add_executable(image_publisher_exe src/node/image_publisher.cpp)
 SET_TARGET_PROPERTIES(image_publisher_exe PROPERTIES OUTPUT_NAME image_publisher)
-target_link_libraries(image_publisher_exe ${catkin_LIBRARIES})
+target_link_libraries(image_publisher_exe ${catkin_LIBRARIES} ${YAML_CPP_LIBRARIES})
 
 install(TARGETS image_publisher_exe
         DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
