--- libs/lz4-r131/cmake_unofficial/CMakeLists.txt
+++ libs/lz4-r131/cmake_unofficial/CMakeLists.txt
@@ -14,7 +14,7 @@
     MESSAGE( STATUS "64 bit architecture detected size of void * is " ${SIZEOF_VOID_P})
 ENDIF()
 
-option(BUILD_TOOLS "Build the command line tools" ON)
+option(BUILD_TOOLS "Build the command line tools" OFF)
 option(BUILD_LIBS  "Build the libraries in addition to the tools" ON)
 
 IF("${CMAKE_C_COMPILER_ID}" STREQUAL "GNU" OR
