--- catkin_ws/src/navigation/map_server/CMakeLists.txt
+++ catkin_ws/src/navigation/map_server/CMakeLists.txt
@@ -12,11 +12,19 @@ find_package(Bullet REQUIRED)
 find_package(SDL REQUIRED)
 find_package(SDL_image REQUIRED)
 
-find_package(PkgConfig REQUIRED)
-pkg_check_modules(YAMLCPP yaml-cpp REQUIRED)
-if(YAMLCPP_VERSION VERSION_GREATER "0.5.0")
+find_package(PkgConfig)
+
+if (ANDROID)
+    find_package(yaml-cpp REQUIRED)
+    add_definitions(-DHAVE_YAMLCPP_GT_0_5_0)
+else()
+  pkg_check_modules(YAML_CPP yaml-cpp REQUIRED)
+  if(YAMLCPP_VERSION VERSION_GREATER "0.5.0")
     add_definitions(-DHAVE_YAMLCPP_GT_0_5_0)
+  endif()
+  link_directories(${YAML_CPP_LIBRARY_DIRS})
 endif()
+include_directories(${YAML_CPP_INCLUDE_DIRS})
 
 catkin_package(
     INCLUDE_DIRS
