--- catkin_ws/src/perception_pcl/pcl_ros/CMakeLists.txt
+++ catkin_ws/src/perception_pcl/pcl_ros/CMakeLists.txt
@@ -181,20 +181,20 @@ class_loader_hide_library_symbols(pcl_ros_surface)
 
 ## Tools
 
-add_executable(pcd_to_pointcloud tools/pcd_to_pointcloud.cpp)
-target_link_libraries(pcd_to_pointcloud ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${EIGEN3_LIBRARIES} ${PCL_LIBRARIES})
+#add_executable(pcd_to_pointcloud tools/pcd_to_pointcloud.cpp)
+#target_link_libraries(pcd_to_pointcloud ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${EIGEN3_LIBRARIES} ${PCL_LIBRARIES})
 
-add_executable(pointcloud_to_pcd tools/pointcloud_to_pcd.cpp)
-target_link_libraries(pointcloud_to_pcd ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${EIGEN3_LIBRARY_DIRS} ${PCL_LIBRARIES})
+#add_executable(pointcloud_to_pcd tools/pointcloud_to_pcd.cpp)
+#target_link_libraries(pointcloud_to_pcd ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${EIGEN3_LIBRARY_DIRS} ${PCL_LIBRARIES})
 
-add_executable(bag_to_pcd tools/bag_to_pcd.cpp)
-target_link_libraries(bag_to_pcd pcl_ros_tf ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${EIGEN3_LIBRARY_DIRS} ${PCL_LIBRARIES})
+#add_executable(bag_to_pcd tools/bag_to_pcd.cpp)
+#target_link_libraries(bag_to_pcd pcl_ros_tf ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${EIGEN3_LIBRARY_DIRS} ${PCL_LIBRARIES})
 
-add_executable(convert_pcd_to_image tools/convert_pcd_to_image.cpp)
-target_link_libraries(convert_pcd_to_image ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${EIGEN3_LIBRARIES} ${PCL_LIBRARIES})
+#add_executable(convert_pcd_to_image tools/convert_pcd_to_image.cpp)
+#target_link_libraries(convert_pcd_to_image ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${EIGEN3_LIBRARIES} ${PCL_LIBRARIES})
 
-add_executable(convert_pointcloud_to_image tools/convert_pointcloud_to_image.cpp)
-target_link_libraries(convert_pointcloud_to_image ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${EIGEN3_LIBRARIES} ${PCL_LIBRARIES})
+#add_executable(convert_pointcloud_to_image tools/convert_pointcloud_to_image.cpp)
+#target_link_libraries(convert_pointcloud_to_image ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${EIGEN3_LIBRARIES} ${PCL_LIBRARIES})
 
 ## Downloads
 
@@ -231,11 +231,11 @@ install(
     pcl_ros_filters
     pcl_ros_surface
     pcl_ros_segmentation
-    pcd_to_pointcloud
-    pointcloud_to_pcd
-    bag_to_pcd
-    convert_pcd_to_image
-    convert_pointcloud_to_image
+#    pcd_to_pointcloud
+#    pointcloud_to_pcd
+#    bag_to_pcd
+#    convert_pcd_to_image
+#    convert_pointcloud_to_image
   RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
   ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
   LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
