--- catkin_ws/src/robot_state_publisher/CMakeLists.txt	2018-04-05 22:22:21.000000000 +0200
+++ catkin_ws/src/robot_state_publisher/CMakeLists.txt	2018-12-05 23:52:33.042000625 +0100
@@ -70,6 +70,7 @@
 endif()
 
 install(TARGETS ${PROJECT_NAME}_solver joint_state_listener ${PROJECT_NAME} state_publisher
+  ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
   LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
   RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION})
 
