--- catkin_ws/src/urdf/urdf/CMakeLists.txt
+++ catkin_ws/src/urdf/urdf/CMakeLists.txt
@@ -26,7 +26,7 @@
 catkin_package(
   LIBRARIES ${PROJECT_NAME}
   INCLUDE_DIRS include ${CATKIN_DEVEL_PREFIX}/include
-  CATKIN_DEPENDS rosconsole_bridge roscpp
+  CATKIN_DEPENDS pluginlib rosconsole_bridge roscpp
   DEPENDS urdfdom_headers urdfdom Boost TinyXML
 )
 install(FILES ${generated_compat_header} DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION})
